WebApp/src/main/java/springboot/WebApp/services/Matcher.java

```diff
@@ -0,0 +1,251 @@
+                tradesList.add(tradedObj);
+                System.out.println("Matched BUY");
+                System.out.println("Matching End");
+                i--;
+                updateOrderBook(newOrder, currentSellOrder);
+                completeTrade(newOrder, currentSellOrder);
+            }
+
+
+        }
+    }
+
+    private void matchSeller(Order newOrder) {
+        for (int i = 0; i < buyOrders.size(); i++) {
+            Order currentBuyOrder = buyOrders.get(i);
+            if (newOrder.getPrice() >= currentBuyOrder.getPrice() && newOrder.getQuantity() != 0) {
+                Trade tradedObj = new Trade(currentBuyOrder, newOrder, currentBuyOrder.getPrice(), Math.min(newOrder.getQuantity(), currentBuyOrder.getQuantity()));
+                tradesList.add(tradedObj);
+                System.out.println("Matched Sell");
+                System.out.println("Matching End");
+                i--;
+                updateOrderBook(currentBuyOrder, newOrder);
+                completeTrade(currentBuyOrder, newOrder);
+
+            }
+
+
+        }
+
+
+    }
+
+
+    private void completeTrade(Order buyOrder, Order sellOrder) {
+        System.out.println("\nBefore Complete Trade\nBuyOrder: " +
+                buyOrder.showData() +
+                "\nSellOrder: " +
+                sellOrder.showData());
+
+        if (buyOrder.getQuantity() > sellOrder.getQuantity()) {
+            buyOrder.setQuantity((buyOrder.getQuantity() - sellOrder.getQuantity()));
+            sellOrder.setQuantity(0);
+        } else {
+            sellOrder.setQuantity((sellOrder.getQuantity()) - buyOrder.getQuantity());
+            buyOrder.setQuantity(0);
+
+        }
+        cleanUp(buyOrder, sellOrder);
+        System.out.println("\nAfter Complete Trade\nBuyOrder: " +
+                buyOrder.showData() +
+                "\nSellOrder: " +
+                sellOrder.showData());
+    }
+
+    private void cleanUp(Order buyOrder, Order sellOrder) {
+
+        if (buyOrder.getQuantity() == 0) {
+            buyOrders.remove(buyOrders.indexOf(buyOrder));
+        }
+        if (sellOrder.getQuantity() == 0) {
+            sellOrders.remove(sellOrders.indexOf(sellOrder));
+        }
+    }
+
+    private void addOrderBookEntry(Order newOrder) {
+        int existingEntryIndex = findExistingIndex(newOrder);
+        if (existingEntryIndex == -1) {
+            OrderBookEntry newEntry = new OrderBookEntry(newOrder.getAction(), newOrder.getPrice(), newOrder.getQuantity());
+            aggOrderBook.add(newEntry);
+            System.out.println("\nOrderBook Added: " + newOrder.showData() + "\n");
+        } else {
+            aggregateOrderBook(newOrder, existingEntryIndex);
+            System.out.println("Order Book Aggregated: ");
+        }
+        Collections.sort(aggOrderBook, new OrderBookSort());
+
+    }
+
+    private int findExistingIndex(Order newOrder) {
```

Comment:
You can leverage on Java's Stream API to get the object. Your code would look like 

aggOrderBook.filter(order -> order.getPrice() == newOrder.getPrice() && order.getAction().equals(newOrder.getActions()).findAny().orElse(null)